### PR TITLE
Prepend event listeners

### LINKED ISSUE #7: Use `emitter.prependOnceListener(eventName, listener)`
See https://github.com/sindresorhus/got/issues/874

Just need to replace every `.once` with `.prependOnceListener` and do some basic test:

```js
test('prepends once listener', t => {
	...
	const timings = timer(request);
	request.once('response' => {
		t.true(typeof timings.response === 'number');
	});
	...
});
```

### PR DESCRIPTION
**Pull request by tobenna**
fixes https://github.com/szmarczak/http-timer/issues/7

Seems to work great but unable to write a failing test.

### CODE REVIEW

File: test.ts [hunk 1]
```diff
@@ -168,3 +168,14 @@ test('sensible timings', async t => {
...
 	t.true(timings.phases.total! < 1000);
 });
+
+test.cb('prepends once listener', t => {
+	const request = https.get('https://httpbin.org/anything')
+
+	const timings = timer(request)
+
+	request.on('response', () => {
+		t.true(typeof timings.response === 'number');
```
Comment (sindresorhus) [inline test.ts]:
```suggestion
		t.is(typeof timings.response, 'number');
```

File: test.ts [hunk 2]
```diff
@@ -168,3 +168,13 @@ test('sensible timings', async t => {
 	t.true(timings.phases.download! < 1000);
 	t.true(timings.phases.total! < 1000);
 });
+
+test.cb('prepends once listener', t => {
+	const request = https.get('https://httpbin.org/anything');
+	const timings = timer(request);
+
+	request.on('response', () => {
```
Comment (sindresorhus) [inline test.ts]: Use `pEvent` like done in the other tests.

File: source/index.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import {EventEmitter} from 'events';
 import {Socket} from 'net';
 import {ClientRequest, IncomingMessage} from 'http';
 // @ts-ignore
-import deferToConnect from 'defer-to-connect';
+import deferToConnect = require('defer-to-connect');
```
Comment (tobenna) [inline source/index.ts:5]: why this?

Comment (szmarczak) [reply source/index.ts:5]: It's not a TS module. @sindresorhus has explained it somewhere but I can't find it now.

**Comment by szmarczak** [comment]
:heart: great job

File: source/index.ts (hunk 3 above)
Comment (szmarczak) [reply source/index.ts:5]: It's also explained here: https://github.com/DefinitelyTyped/DefinitelyTyped/#should-i-add-an-empty-namespace-to-a-package-that-doesnt-export-a-module-to-use-es6-style-imports

Comment (szmarczak) [reply source/index.ts:5]: Feel free to pick up https://github.com/szmarczak/defer-to-connect/issues/2

**Pull request merged**